vault/external_tests/raft/raft_autopilot_test.go

```diff
@@ -403,68 +543,90 @@ func joinAndStabilize(t *testing.T, core *vault.TestClusterCore, client *api.Cli
+// clusterLeader gets the leader node and its address from the specified cluster
+func clusterLeader(t *testing.T, cluster *vault.TestCluster) (*vault.TestClusterCore, string) {
+	for _, core := range cluster.Cores {
+		isLeader, addr, _, err := core.Leader()
+		require.NoError(t, err)
+		if isLeader {
+			return core, addr
+		}
+	}
 
-	client := cluster.Cores[0].Client
+	t.Fatal("unable to find leader")
+	return nil, ""
+}
 
-	config, err := client.Sys().RaftAutopilotConfiguration()
-	require.NoError(t, err)
-	joinAndStabilizeAndPromote(t, cluster.Cores[1], client, cluster, config, "core-1", 2)
-	joinAndStabilizeAndPromote(t, cluster.Cores[2], client, cluster, config, "core-2", 3)
+// setupLeaderAndUnseal configures and unseals the leader node.
+// It will wait until the node is active before returning the core and the address of the leader.
+func setupLeaderAndUnseal(t *testing.T, cluster *vault.TestCluster) (*vault.TestClusterCore, *testhelpers.TestRaftServerAddressProvider) {
+	leader, _ := clusterLeader(t, cluster)
 
-	errIfNonVotersExist := func() error {
-		t.Helper()
-		resp, err := client.Sys().RaftAutopilotState()
-		if err != nil {
-			t.Fatal(err)
+	// Lots of tests seem to do this when they deal with a TestRaftServerAddressProvider, it makes the test work rather than error out.
+	atomic.StoreUint32(&vault.TestingUpdateClusterAddr, 1)
+
+	addressProvider := &testhelpers.TestRaftServerAddressProvider{Cluster: cluster}
+	testhelpers.EnsureCoreSealed(t, leader)
+	leader.UnderlyingRawStorage.(*raft.RaftBackend).SetServerAddressProvider(addressProvider)
+	cluster.UnsealCore(t, leader)
+	vault.TestWaitActive(t, leader.Core)
+
+	return leader, addressProvider
+}
+
+// waitForCoreUnseal waits until the specified core is unsealed.
+// It fails the calling test if the deadline has elapsed and the core is still sealed.
+func waitForCoreUnseal(t *testing.T, core *vault.TestClusterCore) {
+	deadline := time.Now().Add(30 * time.Second)
+	for time.Now().Before(deadline) {
+		if !core.Sealed() {
+			return
 		}
-		for k, v := range resp.Servers {
-			if v.Status == "non-voter" {
-				return fmt.Errorf("node %q is a non-voter", k)
-			}
+		time.Sleep(time.Second)
+	}
+	t.Fatalf("expected core %v to unseal before deadline but it has not", core.NodeID)
+}
+
+// isHealthyAfterStabilization will use the supplied leader core to query the health of Raft Autopilot up until just after the supplied deadline.
+// It will return when the Raft Autopilot state is reported as healthy, or just after the specified stabilization time.
+func isHealthyAfterStabilization(t *testing.T, leaderCore *vault.TestClusterCore, stabilizationTime time.Duration) bool {
+	timeoutGrace := 2 * time.Second
+	deadline := time.Now().Add(stabilizationTime).Add(timeoutGrace)
+	healthy := false
+	for time.Now().Before(deadline) {
+		state, err := leaderCore.Client.Sys().RaftAutopilotState()
+		require.NoError(t, err)
+		if state.Healthy {
+			healthy = true
```

Comment:
We want to ensure that the cluster is healthy *after* the specified stabilization time, if we return before the end of stabilization time then we would still have to ensure we've waited for that time to pass before things like promotion will occur.

Reply:
In that case there's no reason to poll, right?  We could just sleep and then check the state at the end.  Also I think your godoc may need updating:

```
// It will return when the Raft Autopilot state is reported as healthy, or just after the specified stabilization time.
```